source/MaterialXRuntime/RtObject.cpp

```diff
@@ -101,6 +101,16 @@ const RtTypedValue* RtObject::getMetadata(const RtToken& name) const
     return hnd()->asA<PvtObject>()->getMetadata(name);
 }
 
+RtTypedValue* RtObject::getMetadata(const RtToken& name, const RtToken& type)
+{
+    return hnd()->asA<PvtObject>()->getMetadata(name, type);
+}
+
+const RtTypedValue* RtObject::getMetadata(const RtToken& name, const RtToken& type) const
+{
+    return hnd()->asA<PvtObject>()->getMetadata(name, type);
```

Comment:
Is it better to do a const_cast and have one call the other? That way if one implementation changes, they both get changed? Just a thought.

Reply:
You seem to do this sort of thing with all const vs. non-const examples in this file. If you like the suggestion, you can apply it to all such cases.

Reply:
In general I agree with that design. In this case though, since the methods are not inline, it will results in an extra function call being made in the const case. And we can't make the methods inline since they are accessing the private API.

So for performance, in this particular case I prefer to keep the code duplication. It's just a single line and not too bad.